### ISSUE TITLE
getting error with slidify - could not find function regex 

### ISSUE THREAD

**Issue by mb1773** [ramnathv/slidify#430]
I have been through all the help files
installed the correct stringr (stringr_0.6.2.tar.gz)
newest version of rstudio
have slidify and slidify libraries

I have run this exact code (index.Rmd) on my laptop from work with all the same settings (as far as I can tell) and get the slidify but cannot publish to rpubs (SSL error and as it is my work laptop I cannot edit due to admin rights).

So I have installed everything on my home laptop and cannot get to the slides part due to error below.

> setwd("c:/coursera/projdeck")
> slidify("index.Rmd")
Error: could not find function "slidify"
> library("slidify", lib.loc="~/R/win-library/3.2")
> library("slidifyLibraries", lib.loc="~/R/win-library/3.2")
> library("slidifyLibraries", lib.loc="~/R/win-library/3.2")
> slidify("index.Rmd")

processing file: index.Rmd
  |......................                                           |  33%
  ordinary text without R code

  |...........................................                      |  67%
label: unnamed-chunk-1
  |.................................................................| 100%
  ordinary text without R code

output file: index.md

Error in parse_body(x[2]) : could not find function "regex"

**Comment by ramnathv** [comment]
That is a weird error. Can you post your `sessionInfo()` so I can see what might be going on.

**Comment by mb1773** [comment]
Here is the session info...
ignore that first error as I had a typo
in this example I am trying to run the basic index.Rmd with basically no edits
this morning I even uninstalled rStudio and reinstalled to make sure it's a good copy of the program
here is the basic index.Rmd script
---
title       : 
subtitle    : 
author      : 
job         : 
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      # 
widgets     : []            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
knit        : slidify::knit2slides
---

## Read-And-Delete

some plain text to test

--- .class #id 

## Slide 2

and here is the sessionInfo


R version 3.2.0 (2015-04-16) -- "Full of Ingredients"
Copyright (C) 2015 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

[Workspace loaded from ~/.RData]

> library("slidify", lib.loc="~/R/win-library/3.2")
> library("slidifyLibraries", lib.loc="~/R/win-library/3.2")
> setwd("c:/coursera/deck2")
> slidify("index.Rmd")


processing file: index.Rmd
  |.................................................................| 100%
  ordinary text without R code


output file: index.md

Error in yaml.load(gsub("^---\n+", "", txt[1])) : 
  Scanner error: mapping values are not allowed in this context at line 2, column 13
> sessionInfo()
R version 3.2.0 (2015-04-16)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 8 x64 (build 9200)

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] slidifyLibraries_0.3.1 slidify_0.4.5         

loaded via a namespace (and not attached):
[1] markdown_0.7.7  htmltools_0.2.6 tools_3.2.0     whisker_0.3-2   yaml_2.1.13     rmarkdown_0.6.1 knitr_1.10.5   
[8] stringr_0.6.2   digest_0.6.8

**Comment by mb1773** [comment]
I found an old issue resolution and you suggested putting library(knitr) above the first
---
I tried this.
It will now make a HTML file but all the slides are on one page and there is no color or format.

**Comment by FroggyZ** [comment]
I have exactly the same issue.

here is my sessionInfo():

R version 3.2.0 (2015-04-16)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=French_Canada.1252  LC_CTYPE=French_Canada.1252    LC_MONETARY=French_Canada.1252 LC_NUMERIC=C                  
[5] LC_TIME=French_Canada.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.10.5   slidify_0.4.5  devtools_1.8.0

loaded via a namespace (and not attached):
 [1] XML_3.98-1.1    digest_0.6.8    bitops_1.0-6    jsonlite_0.9.14 formatR_1.2     git2r_0.10.1    httr_0.6.1      whisker_0.3-2  
 [9] rmarkdown_0.6.1 tools_3.2.0     stringr_1.0.0   RCurl_1.95-4.6  markdown_0.7.7  yaml_2.1.13     rversions_1.0.0 memoise_0.2.1  
[17] htmltools_0.2.6

**Comment by ramnathv** [comment]
This should work. Run the following before loading `slidify` and reknitting your document. This installs the latest version of `stringr` and also the updates from @muschellij2 that fixes the parser for changes in `stringr`. I just tested it and it works beautifully. I will be merging these updates into the master, but for now this should do the job.

```r
install.packages('stringr')
devtools::install_github('muschellij2/slidify')
```

I would appreciate if you can try this out and report back.

**Comment by FroggyZ** [comment]
Thanks a lot ramnathv. I 'll try as soon as I can and let you know.

Le 2015-05-26 21:40, "Ramnath Vaidyanathan" <notifications@github.com> a
écrit :
>
> This should work. Run the following before loading slidify and reknitting
your document. This installs the latest version of stringr and also the
updates from @muschellij2 that fixes the parser for changes in stringr. I
just tested it and it works beautifully. I will be merging these updates
into the master, but for now this should do the job.

**Comment by janeoeyre** [comment]
@ramnathv It still doesn't work. If performed the following actions: install.packages('stringr')
devtools::install_github('muschellij2/slidify'). The error messages "perl is deprecated. Please use regexp instead" will appear. 
If reinstall the stringr to 0.6.2 version, then the error message "Error in parse_body(x[2]) : could not find function "regex"" will appear again. What is the solution?

**Comment by ramnathv** [comment]
What is your `sessionInfo()`? It seems like you haven't updated some of your packages.

**Comment by janeoeyre** [comment]
My sessionInfo() is following:
R version 3.1.2 (2014-10-31)
Platform: x86_64-apple-darwin13.4.0 (64-bit)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] slidify_0.4.5  devtools_1.8.0

loaded via a namespace (and not attached):
 [1] bitops_1.0-6    digest_0.6.8    formatR_1.2     git2r_0.10.1    knitr_1.10.5   
 [6] markdown_0.7.7  memoise_0.2.1   RCurl_1.95-4.6  rversions_1.0.0 stringr_0.6.2  
[11] tools_3.1.2     whisker_0.3-2   XML_3.98-1.2    yaml_2.1.13

**Comment by ramnathv** [comment]
You still have `stringr_0.6.2`. Try this sequence.

```r
install.package('stringr')
if (pacakgeVersion('stringr') > '1.0.0'){
 devtools::install_github('muschellij2/slidify')
} else {
  cat('You need stringr 1.0.0'
}
```

**Comment by fmahdi-R** [comment]
Hi Ramnath, I got the same error despite having 1.0.0 version of stringr. 

> packageVersion('stringr')
> [1] ‘1.0.0’
> library("stringr")
> Error in unloadNamespace(package) : 
>   namespace ‘stringr’ is imported by ‘swirl’, ‘reshape2’, ‘httr’ so cannot be unloaded
> Error in library("stringr") : “stringr” version 0.6.2 cannot be unloaded.

So it seems I needed to clear my R session since the older version of 'stringr' was loaded. Hope this helps some other users.

**Comment by ramnathv** [comment]
Yes. Installing a new version of a package while it is already loaded can cause issues intermittently. It is always advisable to reload your R session when you install new packages.

**Comment by akhilapant** [comment]
Hi ramnathv

I am facing the same problem while running slidify command. I see stringr 0.6.2 version in my sessioninfo(). However, I am not successful installing 1.0 version of stringr.  Looking forward to your help

> slidify('index.Rmd')

processing file: index.Rmd
  |.................................................................| 100%
  ordinary text without R code

output file: index.md

Error in parse_body(x[2]) : could not find function "regex"

Akhila Pant
